Orange/widgets/data/owcreateclass.py

```diff
@@ -90,26 +134,14 @@ def transform(self, c):
         c = c.astype(str)
         c[nans] = ""
         res = map_by_substring(
-            c, self.patterns, self.case_sensitive, self.match_beginning,
+            c, self.patterns,
+            self.case_sensitive, self.match_beginning, self.regular_expressions,
             self.map_values)
         res[nans] = np.nan
         return res
 
-    def __eq__(self, other):
-        return super().__eq__(other) \
-               and self.patterns == other.patterns \
-               and self.case_sensitive == other.case_sensitive \
-               and self.match_beginning == other.match_beginning \
-               and self.map_values == other.map_values
 
-    def __hash__(self):
-        return hash((type(self), self.variable,
-                     tuple(self.patterns),
-                     self.case_sensitive, self.match_beginning,
-                     self.map_values))
-
-
-class ValueFromDiscreteSubstring(Lookup):
+class ValueFromDiscreteSubstring(Lookup, _EqHashMixin):
```

Comment:
I suppose this should be the other way around:
`ValueFromDiscreteSubstring(_EqHashMixin, Lookup)`